apis/v1/shared_types.go

```diff
@@ -855,3 +855,15 @@ const (
 	// Support: Extended
 	PermanentCookieLifetimeType CookieLifetimeType = "Permanent"
 )
+
+type Fraction struct {
```

Comment:
Recommend adding CEL validation at this level that ensures that `numerator <= denominator`, something similar to: https://github.com/kubernetes-sigs/gateway-api/blob/8f5f9d1f515bfd1c2ac9eeee32e436807e0c43ff/apis/v1/httproute_types.go#L312